python/tvm/relax/block_builder.py

```diff
@@ -661,12 +661,20 @@ def normalize(self, expr: Expr) -> Expr:
     def get(self) -> tvm.IRModule:
         """Return the IRModule being built.
 
+        Note this call may invalidate global vars previously returned by this builder
+        (see tvm.relax.transform.NormalizeGlobalVar), so it is recommended to call this function
+        once at the end of the build process.
+
         Returns
         -------
         ret : tvm.IRModule
             An IRModule with Relax and TIR functions being built.
         """
-        return _ffi_api.BlockBuilderGetContextIRModule(self)  # type: ignore
+        # avoid circular import
+        from .transform import NormalizeGlobalVar
+
+        mod = _ffi_api.BlockBuilderGetContextIRModule(self)  # type: ignore
```

Comment:
Is it possible to embed the pass inside the `BlockBuilderGetContextIRModule`? aka we can have the same behavior on both python and cpp side

Reply:
We have considered that. Here the problem is NormalizeGlobalVar will invalidate all global vars previously acquired. But I cannot ensure all pass will follow the assumption that no global var will be used again after BlockBuilderGetContextIRModule. Actually, some passes like AllocateWorkspace might already break the assumption.

On the other hand, on python side BlockBuilder is mainly to build a new IRModule from scratch, and not involve too complex logic. So apply NormalizeGlobalVar to bb.get() could be a good choice.

Reply:
I discussed it with @Ubospica and think that we should not turn it on by default. Because there are not too many cases we create new public functions, and renormlaization can be costly. We should however add wellform check to ensure normal passes do not generate code that violate this assumption


Maybe we do not make it part of `BlockBuilder.get()` but instead add a new function `BlockBuilder.finalize()` and add remark that it will also normalizes the calls.

Reply:
But I think this will make it confused to use `bb.finalize()` or `bb.get()`, because their only difference is whether apply NormalizeGlobalVar or not. If we require users only use `bb.finalize()` at the end of the IRModule building process, that means a major behaviour change and all related tests might need to be changed.

Reply:
OK, we can keep this for now and keep the documentaiton.